authors/apps/profiles/tests/test_folow.py

```diff
@@ -0,0 +1,108 @@
+from .test_data import valid_user, valid_user2
+
+class ProfileFollowAPIViewTestCase(TestCase):
+    """
+    This class defines the test suite for the view
+    that retrieves and updates a user
+    """
+
+    def setUp(self):        
+        user1 = User.objects.create_user(**valid_user)
+        user2 = User.objects.create_user(**valid_user2)
+        user2.is_active = True
+        user2.save()
+
+        self.client = APIClient()
+        self.client.force_authenticate(user=user1)
+
+
+    def test_un_authenticated_user_cannot_follow_or_unfollow_others(self):
+        self.client.logout()
+        response = self.client.post(
+            '/api/profiles/peter/follow/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_un_authenticated_user_cannot_manage_followers(self):
+        self.client.logout()
+        response = self.client.get(
+            '/api/profiles/peter/following/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_authenticated_user_cannot_follow_unknown_profile(self):
+        response = self.client.post(
+            '/api/profiles/fred/follow/'
+        )
+        self.assertIn(
+            "The requested profile does not exist.",
+            response.data["errors"]["detail"])
+
+    def test_authenticated_user_can_follow_another_user(self):
+        response = self.client.post(
+            '/api/profiles/peter/follow/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_authenticated_user_can_re_follow_same_user(self):
+        response = self.client.post(
+            '/api/profiles/peter/follow/'
+        )
+
+        response = self.client.post(
+            '/api/profiles/peter/follow/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_authenticated_user_can_themselves(self):
+        response = self.client.post(
+            '/api/profiles/henryjones/follow/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_auth_user_can_unfollow_users_they_follow(self):
+        response = self.client.post(
+            '/api/profiles/peter/follow/'
+        )
+        response = self.client.delete(
+            '/api/profiles/peter/follow/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_auth_user_cannot_unfollow_users_they_are_not_following(self):
+        response = self.client.delete(
+            '/api/profiles/peter/follow/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_auth_user_cannot_unfollow_non_present_users(self):
+        response = self.client.delete(
+            '/api/profiles/fred/follow/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_auth_user_see_followers_of_invalid_account(self):
```

Comment:
On the method name here, I am not sure what it is you are testing. Are you testing whether a user **cannot** see followers or that they **can** see followers? Note that this issue is the same for the method names on lines 95 and 104.

Reply:
This is also noted, let me update it.